aten/src/ATen/core/Allocator.h

```diff
@@ -119,6 +119,12 @@ struct Allocator {
     AT_ASSERT(d);
     d(ptr);
   }
+  virtual void prepare(size_t s) {
```

Comment:
These need docs. What are they supposed to do? Why do they have to be virtual methods? (Note that you're always free to define extra methods on the subclass and call them directly; it's only necessary to put something as a virtual method on the parent class if you need to call them in a situation where you **don't** know what the concrete implementation is. So maybe you don't even need these at all!)